tx_check.go

```diff
@@ -27,11 +27,14 @@ func (tx *Tx) Check(options ...CheckOption) <-chan error {
 	}
 
 	ch := make(chan error)
-	go tx.check(chkConfig.kvStringer, ch)
+	go tx.check(chkConfig, ch)
 	return ch
 }
 
-func (tx *Tx) check(kvStringer KVStringer, ch chan error) {
+func (tx *Tx) check(cfg checkConfig, ch chan error) {
+	// Close the channel to signal completion.
+	defer close(ch)
```

Comment:
IMO, closing of channel should be the responsibility of caller.

Reply:
sort of agreeing on this, but not a big deal. 

Let's take care of in a followup PR with change in the method `Check` something like below,

```
go func() {
    defer close(ch)
    tx.check(chkConfig, ch)
}()
```

Reply:
Raised a separate PR https://github.com/etcd-io/bbolt/pull/666